contract_membership_delegated_partner/tests/test_delegated_partner_contract.py

```diff
@@ -9,7 +9,13 @@ class TestMembershipDelegateSetup(TestContractBase):
     def setUpClass(cls):
         super(TestMembershipDelegateSetup, cls).setUpClass()
         cls.partner2 = cls.env["res.partner"].create({"name": "Mrs. Odoo"})
-        cls.product_1.membership = True
+        cls.product_1.write(
+            {
+                "membership_date_from": "2023-01-01",
```

Comment:
Why do you need to change this?

Reply:
@pedrobaeza Because the new constraint won't let a membership product have no start/end dates.

```python
    @api.constrains("membership_date_from", "membership_date_to", "membership")
    def _check_membership_dates(self):
        if self.filtered(
            lambda record: record.membership
            and (not record.membership_date_from or not record.membership_date_to)
        ):
            raise ValidationError(
                _("A membership product must have a start date and an end date.")
            )
```

(Interestingly, the write function is sensitive to the order of key-value pairs. If `"membership": True` appears before the other items, the constraint complains. Not sure why that happens.)

Reply:
It's weird, as in other places, you are removing such fields, but it's not a big deal.

/ocabot merge patch